src/pytest_recording/network.py

```diff
@@ -75,34 +94,46 @@ def uninstall_pycurl_wrapper():
     sys.modules["pycurl"] = pycurl
 
 
-def block_pycurl():
+def block_pycurl(allowed_hosts=None):
     global _disable_pycurl  # pylint: disable=global-statement
+    global _allowed_hosts  # pylint: disable=global-statement
     _disable_pycurl = True
+    _allowed_hosts = allowed_hosts
 
 
 def unblock_pycurl():
     global _disable_pycurl  # pylint: disable=global-statement
+    global _allowed_hosts  # pylint: disable=global-statement
     _disable_pycurl = False
+    _allowed_hosts = None
 
 
-def block_socket():
-    socket.socket.connect = network_guard
-    socket.socket.connect_ex = network_guard
+def block_socket(allowed_hosts=None):
+    socket.socket.connect = make_network_guard(_original_connect, allowed_hosts=allowed_hosts)
+    socket.socket.connect_ex = make_network_guard(_original_connect_ex, allowed_hosts=allowed_hosts)
 
 
 def unblock_socket():
     socket.socket.connect = _original_connect
     socket.socket.connect_ex = _original_connect_ex
 
 
-def network_guard(*args, **kwargs):
-    raise RuntimeError("Network is disabled")
+def make_network_guard(original_func, allowed_hosts=None):
+    def network_guard(*args, **kwargs):
```

Comment:
Please, add `functools.wraps` decorator

Reply:
It will make `__name__` of the guard function equal to `socket.socket.connect` and a lot of checks in tests will fail.

Reply:
aha, ok